internal/common/repo.go

```diff
@@ -0,0 +1,25 @@
+package common
+
+import (
+	"encoding/json"
+
+	"github.com/razorpay/metro/pkg/registry"
+)
+
+const (
+	basePrefix = "registry/"
+)
+
+// Repo implements base repo methods
+type Repo struct {
+	Registry registry.Registry
+}
+
+// Create puts the model in the registry
+func (r Repo) Create(m Model) error {
+	b, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+	return r.Registry.Put(basePrefix+m.Prefix()+m.Key(), b)
```

Comment:
should this be exposed by a method in model ?
getKey(){
return basePrefix+m.Prefix()+m.Key()
}

Reply:
I think prefix and key can come from Key() method, but base prefix can be in base class itself?

Reply:
on the other hand, for list functionality, I just need basePrefix+m.Prefix(), so the current scheme works there. Thoughts?